apps/bbsync/query.py

```diff
@@ -411,7 +411,7 @@ def generate_comment(self):
         if pending_comments.exists():
             self._query["comment"] = {
                 "body": pending_comments.first().text,
-                "is_private": False,
+                "is_private": pending_comments.first().is_private,
```

Comment:
I think this is not atomic and could potentially lead to a mismatch (text from one comment and is_private flag from another one) when new pending comment is created between the two `.first()` calls as Django queries are lazy.

Example with affects:
```python
In [31]: skynet_affects = Affect.objects.filter(ps_component__contains="skynet")

In [32]: skynet_affects.first()
Out[32]: <Affect: c6c31ad6-d57d-4419-9ccd-48474594ac05>

In [33]: # new affect with "skynet" ps_component created

In [34]: skynet_affects.first()
Out[34]: <Affect: 2e2dd85f-181a-4742-bd63-00d5b3142b84>
```

Querying for the comment just once should solve the problem:
```python
comment = pending_comments.first()
self._query["comments"] = {
...
```

Reply:
BBSync is currently triggered by an API call and Django should ensure atomic behavior but if we can make it not relying on that it would be better as it should soon become an async action triggered by backend worker.

Reply:
Fixed, thanks.

Just a note, if this is turned into an async action without the requisite cleanup of pending comments (now done synchronously during import from bugzilla), it will send seemingly random comments to bugzilla, because it just selects one of the pending ones.

Reply:
Oh :smile: that is probably not what would people expect but at least it would be never boring.